contracts/index_executor/src/steps/mod.rs

```diff
@@ -27,6 +37,12 @@ pub struct Step {
 }
 
 impl Runner for Step {
+    /// @nonce: worker's nonce
+    /// where is worker?
+    /// TODO: worker is in the context, therefore nonce should be in the context
```

Comment:
nonce is distributed for each step when initializing, I think putting nonce along with each step is good, why do you think it should be in the context?